xt_layer7 memory leak fix. Memory for pattern will be allocated by regexcmp library function.

diff --git a/net/netfilter/xt_layer7.c b/net/netfilter/xt_layer7.c
--- a/net/netfilter/xt_layer7.c
+++ b/net/netfilter/xt_layer7.c
@@ -173,15 +173,15 @@ static regexp * compile_and_cache(const 
 	}
 
 	tmp->regex_string  = kmalloc(strlen(regex_string) + 1, GFP_ATOMIC);
-	tmp->pattern       = kmalloc(sizeof(struct regexp),    GFP_ATOMIC);
+//	tmp->pattern       = kmalloc(sizeof(struct regexp),    GFP_ATOMIC);
 	tmp->next = NULL;
 
-	if(!tmp->regex_string || !tmp->pattern) {
+	if(!tmp->regex_string) {
 		if (net_ratelimit())
 			printk(KERN_ERR "layer7: out of memory in "
 					"compile_and_cache, bailing.\n");
 		kfree(tmp->regex_string);
-		kfree(tmp->pattern);
+//		kfree(tmp->pattern);
 		kfree(tmp);
 		return NULL;
 	}
